docs/kuneiform/dml.md

```diff
@@ -30,29 +30,31 @@ action <your_desired_name>() <privacy> {}
 
 ## Defining Action Parameters
 
-Action parameters represent user inputs provided when the action is called. These parameters should be prefixed with a $ symbol, which subsequently maps to the values passed to the SQL query:
+Action parameters represent user inputs provided when the action is called. These parameters should be prefixed with a `$` symbol, which subsequently maps to the values passed to the SQL query:
 
 ```typescript
 action add_user($id, $name, $username) <public | private> {}
 ```
 
 ## Setting Action Privacy
 
-Privacy levels can be set for your action. Similar to public / private in [TypeScript](https://www.tutorialsteacher.com/typescript/data-modifiers), private actions can only be called by other actions in the Kuneiform schema. Public actions can be called by any external database user.
+Privacy levels can be set for your action using [access tags](./supported-features#access-tags). Similar to public / private in [TypeScript](https://www.tutorialsteacher.com/typescript/data-modifiers), private actions can only be called by other actions in the Kuneiform schema. Public actions can be called by any external database user.  Other modifiers can be added to enforce other restrictions on the action.  In the below example, we add an `owner` modifier, which makes the action only callable by the database deployer
+
+A full list of supported privacy levels can be found [here](./supported-features#access-tags).
 
 ```typescript
-action add_user($id, $name, $username) public {}
+action add_user($id, $name, $username) public owner {}
 ```
 
 ## Defining SQL Queries in an Action
 
-The SQL queries to be executed when the action is called can be defined within the action's body. Parameters within the SQL query must be prefixed with $.  Multiple queries can be included in the same action.  The two queries will be executed within one transaction, with either all succeeding or all failing. All query statements **must** end with a semicolon. If queries return results, the final query statement will be returned to the client.
+The SQL queries to be executed when the action is called can be defined within the action's body. Parameters within the SQL query must be prefixed with `$`.  Multiple queries can be included in the same action.  The two queries will be executed within one transaction, with either all succeeding or all failing. All query statements **must** end with a semicolon. If queries return results, the final query statement will be returned to the client.
```

Comment:
I don't think so, `view` should only have meaning regarding to `mutability`, not about returning data.

Reply:
Hmm, but isn't that confusing because broadcast doesn't return data?

Currently, broadcast does not return data. So the only way we can return data an action is with the call endpoint, which would require a `view` action.

Reply:
Oh sorry I misinterpret your comments.  Yes we should recommend to create a `view` action to return data. 

P.S. I think we probably need some change on engine so that non-view action should only return curtain response(maybe just a boolean).

`broadcast` can return data, we haven't return anything yet(except the `log` filed, it'll be 'success' if tx succeed).
The problem for returning data from `broadcast` is that everything returned(data, log, events,etc) will be stored on chain, it's not suitable to return a big chunk of data.

Reply:
Yeah, that makes sense.

It seems like the most limiting part of this system (state-changes got to broadcast, read-queries go to call) is that we can no longer have actions that change a bunch of state, then return data at the end. Just to confirm, that is not currently possible in our system, correct? Is it even possible to enable?

Reply:
Current implementation won't return result from the SQL execution.    
It's possible(that's exactly what we did before we start integrate cometbft), although I don't think we should